integration-tests/testsetups/ocr.go

```diff
@@ -824,70 +823,106 @@ func (o *OCRSoakTest) setFilterQuery() {
 		Msg("Filter Query Set")
 }
 
-// observeOCREvents subscribes to OCR events and logs them to the test logger
-// WARNING: Should only be used for observation and logging. This is not a reliable way to collect events.
-func (o *OCRSoakTest) observeOCREvents() error {
-	eventLogs := make(chan types.Log)
-	ctx, cancel := context.WithTimeout(testcontext.Get(o.t), 5*time.Second)
-	eventSub, err := o.seth.Client.SubscribeFilterLogs(ctx, o.filterQuery, eventLogs)
-	cancel()
-	if err != nil {
-		return err
-	}
+// pollingOCREvents Polls the blocks for OCR events and logs them to the test logger
+func (o *OCRSoakTest) pollingOCREvents(endTest <-chan time.Time) error {
+	// Keep track of the last processed block number
+	startingBlockNum := o.startingBlockNum
+	// Initialized to an invalid block number (max value of uint64)
+	lastCheckedBlockNum := ^uint64(0)
 
 	go func() {
+		// TODO: Make this configurable
+		pollInterval := time.Second * 30
+		ticker := time.NewTicker(pollInterval)
+		defer ticker.Stop()
+		o.log.Info().Msg("Start Polling for Answer Updated Events")
 		for {
 			select {
-			case event := <-eventLogs:
```

Comment:
Do you think its worth keeping the WSS option also?

Reply:
I would move onto polling. Seems more reliable and not prone to disconnections.